### PR TITLE
[Parser] Make parser aware of the unexpected attributes in operator declarations

### PR DESCRIPTION
**Pull request by StevenWong12**
To sync the changes in https://github.com/apple/swift-syntax/pull/1448#pullrequestreview-1383999927

### CODE REVIEW

File: lib/Parse/ParseDecl.cpp [hunk 1]
```diff
@@ -4606,6 +4606,14 @@ static void diagnoseOperatorFixityAttributes(Parser &P,
     if (fixityAttrs.empty()) {
       P.diagnose(OD->getOperatorLoc(), diag::operator_decl_no_fixity);
     }
+
+    for (auto it = Attrs.begin(); it != Attrs.end(); ++it) {
+      if (isFixityAttr(*it)) {
+        continue;
+      }
+      auto *attr = *it;
+      P.diagnose(attr->getLocation(), diag::operator_decl_should_not_contain_other_attributes, attr->getAttrName());
```
Comment (ahoppen) [inline lib/Parse/ParseDecl.cpp:4615]: Could you add a Fix-It to this diagnostic to remove the attribute (similar to line 4599).

---

Could you format the line to only contain 80 columns? I usually run `git-clang-format`, which (IIRC) you can install using Homebrew.

File: test/Parse/operator_decl.swift [hunk 2]
```diff
@@ -117,3 +117,5 @@ infix operator *<*< : F, Proto
 
 // expected-error@+2 {{expected precedence group name after ':' in operator declaration}}
 postfix operator ++: // expected-error {{only infix operators may declare a precedence}} {{20-21=}}
+
+dynamic prefix operator +!+  // expected-error{{unexpected attribute 'dynamic' in operator declaration}}
```
Comment (ahoppen) [inline test/Parse/operator_decl.swift:121]: I see that you moved this test case. Could you keep the assertion that there should be a Fix-It to remove `dynamic`, i.e. the `{{1-9=}} part?

---

Could you add a newline at the end of the file?

**Review by ahoppen — COMMENTED** [review]
Thanks, this is great work. Sorry for not providing you pointers earlier, I didn’t have a compiler build locally at the end of last week that I could have used to find the places to change but you figured this out really well on your own 🤩

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Comment by StevenWong12** [comment]
Hey @ahoppen, sorry for the late reply.

I've modified my codes according to your suggestions. 

When I rebuild the stdlib locally, I got some errors as follows
```
<unknown>:0: error: unexpected attribute <<raw doc comment>> in operator declaration
```

It seems that some files used to build the stdlib use the ``<<raw doc comment>>`` as a ``DeclAttribute`` and the log indicates the latest building file is ``SIMDVectorTypes.swift``, but I didn't see any inappropriate use of operator declaration there. Is there anything I missed? If not, maybe I should debug the building stage to locate the problem?

**Comment by ahoppen** [comment]
I’m not 100% sure what’s going wrong here but my bet right now is that you will be able to reproduce the issue with

```swift
/// some doc comment
prefix operator +!+  
```

The compiler models doc comments as attributes and you are marking them as invalid. So, you should also be allowing doc comments (with attribute kind `DAK_RawDocComment`) for operator declarations.

**Comment by StevenWong12** [comment]
I've added doc comments as the allowable attribute in operator declaration.

> my bet right now is that you will be able to reproduce the issue with

But I can not produce this problem locally.

**Comment by ahoppen** [comment]
Let’s see what CI has to say about it.

@swift-ci Please smoke test

**Comment by ahoppen** [comment]
Sorry, I forgot about this PR after I triggered CI. Merged now.

**Pull request merged**